src/helper/horizon-rpc.ts

```diff
@@ -229,3 +229,34 @@ export const fetchAccountHistory = async (
     throw new Error(JSON.stringify(error));
   }
 };
+
+export const submitTransaction = async (
+  signedXDR: string,
+  networkUrl: string,
+  networkPassphrase: string
+): Promise<{
+  data: Horizon.HorizonApi.SubmitTransactionResponse | null;
+  error: unknown;
+}> => {
+  const tx = TransactionBuilder.fromXDR(signedXDR, networkPassphrase);
+  const server = new Horizon.Server(networkUrl);
+
+  try {
+    const data = await server.submitTransaction(tx);
+    return {
+      data,
+      error: null,
+    };
+  } catch (e: any) {
```

Comment:
I know you're just picking up this code from the Freighter codebase, but maybe we can type this error now. I think it should be `BadResponseError` from stellar-sdk

Reply:
oh yeah so I actually tried this but it turns out in TS, you can only annotate the error from the catch block as an `any` or `unknown`. https://byby.dev/ts-try-catch-error-type#:~:text=The%20only%20type%20annotations%20that,on%20it%20without%20type%20checking.

So I did annotate it in the response shape from this function but not sure if I can actually annotate the error as it come in from the catch block.